pkg/util/dump/dump.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dump
+
+import (
+	"github.com/davecgh/go-spew/spew"
+)
+
+var prettyPrintConfig = &spew.ConfigState{
+	Indent:                  "  ",
+	DisableMethods:          true,
+	DisablePointerAddresses: true,
+	DisableCapacities:       true,
+}
+
+var prettyPrintConfigForHash = &spew.ConfigState{
+	Indent:         " ",
+	SortKeys:       true,
+	DisableMethods: true,
+	SpewKeys:       true,
+}
+
+// PrettyPrintObject wrap the spew.Sdump with Indent, and disabled methods like error() and String()
+// The output may change over time, so for guaranteed output please take more direct control
+func PrettyPrintObject(a interface{}) string {
+	return prettyPrintConfig.Sdump(a)
```

Comment:
I had a comment in utils: https://github.com/kubernetes/utils/pull/226#issuecomment-1055356413

In short, we use both `%#v` and `Sdump` in k/k, and `Sdump` has pretty format.

Reply:
Maybe we should expose `dump.ForHash(obj)`, `dump.OneLine(obj)` and `dump.Pretty(obj)` ?

Reply:
Added one for `OneLine`

Reply:
Since we're in code freeze and this can't merge, I'm going to nit-pick.

Let's call this just `Pretty` or `ForHumans`, and the others `ForHash` and `OneLine` - remember that the package is part of the name, so `dump.Pretty` is just as descriptive as `dump.PrettyPrintObject`.